File: queues/ingress/Ruby/lib/result.rb
# frozen_string_literal: true

# Copyright Core DF — Apache License 2.0

module CoreautoResult
  module_function

  def missing_env(vars)
    { status_code: 601, error: "Environment variables #{vars} should be defined" }
  end

  def transport_error(message = 'inaccessible')
    { status_code: 0, error: message }
  end
end

File: queues/ingress/Rust/src/result.rs
// Copyright Core DF — Apache License 2.0

use serde_json::{json, Value};
use std::env;

pub fn missing_env(vars: &str) -> Value {
    json!({
        "status_code": 601,
        "error": format!("Environment variables {vars} should be defined")
    })
}

pub fn transport_error(message: &str) -> Value {
    json!({ "status_code": 0, "error": message })
}
